packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -75,10 +77,24 @@ function preload(
       seen[dep] = true
       const isCss = dep.endsWith('.css')
       const cssSelector = isCss ? '[rel="stylesheet"]' : ''
-      // @ts-ignore check if the file is already preloaded by SSR markup
-      if (document.querySelector(`link[href="${dep}"]${cssSelector}`)) {
+      const isBaseRelative = !!importerUrl
+
+      // check if the file is already preloaded by SSR markup
+      if (isBaseRelative) {
+        // When isBaseRelative is true then we have `importerUrl` and `dep` is
+        // already converted to an absolute URL by the `assetsURL` function
+        for (let i = links.length - 1; i >= 0; i--) {
+          const link = links[i]
+          // The `links[i].href` is an absolute URL thanks to browser doing the work
+          // for us. See https://html.spec.whatwg.org/multipage/common-dom-interfaces.html#reflecting-content-attributes-in-idl-attributes:idl-domstring-5
+          if (link.href === dep && (!isCss || link.rel === 'stylesheet')) {
+            return
+          }
+        }
+      } else if (document.querySelector(`link[href="${dep}"]${cssSelector}`)) {
         return
       }
```

Comment:
Maybe @sapphi-red suggestion [here](https://github.com/vitejs/vite/pull/9970#discussion_r969505865) was to do my initial thought, just in a way that actually fix my issue?
As far as I can tell, it omits the `if (isBaseRelative)`, and the suggestion is to do this:
```suggestion
      const absoluteDep = new URL(dep, document.baseURI).href;
      for (let i = links.length - 1; i >= 0; i--) {
        const link = links[i]
        // The `links[i].href` is an absolute URL thanks to browser doing the work
        // for us. See https://html.spec.whatwg.org/multipage/common-dom-interfaces.html#reflecting-content-attributes-in-idl-attributes:idl-domstring-5
        if (link.href === absoluteDep && (!isCss || link.rel === 'stylesheet')) {
          return
        }
      }
```